File: backend/src/api/v1/endpoints/profiles.py
from fastapi import APIRouter

from src.api import SessionDependency, NoAuthDependency, CurrentUser
from src.schemas import Profiles, ProfileUpdate
from src.services import profiles

router = APIRouter()


@router.get("")
async def get_all(db: NoAuthDependency) -> list[Profiles] | None:
    return await profiles.get_all(db=db)


@router.get("/{id}")
async def get_by_id(db: NoAuthDependency, id: str, ) -> Profiles:
    return await profiles.get(db=db, id=id)


@router.put("/{id}")
async def update_profile(db: SessionDependency, profile_in: ProfileUpdate, user: CurrentUser) -> Profiles:
    return await profiles.update(db=db, obj_in=profile_in, id=user.id)


@router.post("/delete/{id}")
async def delete_profile(db: SessionDependency, profile_id: int) -> Profiles:
    return await profiles.delete(db=db, id=profile_id)

File: client/src/services/profiles.js
import { httpClient } from '../utils/requests'

export const profileRequest = {
  getAll: async () => {
    const response = await httpClient.get('/profiles')

    return response
  },
  getById: async (id) => {
    const response = await httpClient.get(`/profiles/${id}`)
    return response
  },
  create: async (post) => {
    const response = await httpClient.post('/profiles', post)
    return response
  },
  update: async (id, post) => {
    const response = await httpClient.put(`/profiles/${id}`, post)
    return response
  },
  delete: async (id) => {
    const response = await httpClient.delete(`/profiles/${id}`)
    return response
  },
}
